header.py

```diff
@@ -7,4 +9,22 @@ def __init__(self, content, level = 1):
         self._level = level
     
     def render(self):
-        return "<h{0}> {1} </h{0}>".format(self._level, self.content())
+        return "<h{0}>{1}</h{0}>".format(self._level, self.content())
+
+
+
+def parser(content: typing.Text) -> (int, int, Header):
+    header_pattern = r"^([ ]*)(#+)([ ]*)(.+)$"
```

Comment:
the amount of `#` is at most 6 to be `h6`. so that should be `[#]{1-6}` rather than `#+`

Reply:
Intended behavior, using {1-6} will parse ######## as h6 and content with 2 #.